plugin/storage/es/spanstore/writer.go

```diff
@@ -99,12 +102,15 @@ func (s *SpanWriter) CreateTemplates(spanTemplate, serviceTemplate, indexPrefix
 // spanAndServiceIndexFn returns names of span and service indices
 type spanAndServiceIndexFn func(spanTime time.Time) (string, string)
 
-func getSpanAndServiceIndexFn(archive, useReadWriteAliases bool, prefix, spanDateLayout string, serviceDateLayout string) spanAndServiceIndexFn {
-	if prefix != "" {
-		prefix += indexPrefixSeparator
+func getSpanAndServiceIndexFn(archive, useReadWriteAliases bool, spanIndexOpts, serviceIndexOpts cfg.IndexOptions) spanAndServiceIndexFn {
+	if spanIndexOpts.Prefix != "" {
```

Comment:
similar to above
```suggestion
	if spanIndexOpts.Prefix != "" && !strings.HasSuffix(spanIndexOpts.Prefix, indexPrefixSeparator) {
```

Reply:
since we do this in multiple places I would extract into a single function. Ideally these kinds of adjustments should be done very early after parsing the config, not spread out across many places in the code. It could be even a helper function on the IndexOptions struct, e.g.
```
func (o IndexOptions) FullIndexName(name string) {
  if o.Prefix == "" {
    return name
  }
  if strings.HasSuffix(o.Prefix. indexPrefixSeparator) {
    return o.Prefix + name
  }
  return o.Prefix + indexPrefixSeparator + name
}
```